src/raster/processing/dem/options.rs

```diff
@@ -0,0 +1,172 @@
+use std::fmt::{Display, Formatter};
+use std::marker::PhantomData;
+use std::ptr;
+use std::ptr::NonNull;
+
+use gdal_sys::{
+    GDALDEMProcessingOptions, GDALDEMProcessingOptionsFree, GDALDEMProcessingOptionsNew,
+};
+
+use crate::cpl::CslStringList;
+use crate::errors;
+use crate::utils::_last_null_pointer_err;
+
+/// Payload for [`GDALDEMProcessing`]. Intended for internal use only.
+pub struct GdalDEMProcessingOptions<'opts>(
```

Comment:
I didn't mark this as `pub(crate)` in case someone else needed it for a missed binding on a new feature, but maybe it should be?

Reply:
I think it's fine to make it public, but are you sure it needs the lifetime? Looking at the [GDAL code](https://github.com/OSGeo/gdal/blob/59eaaed3168f49e8a7a3821730277aff68a86d16/apps/gdaldem_lib.cpp#L3770), it seems to copy what it needs out of `papszArgv`. So we can probably drop the `PhantomData` and `'opts`.

Reply:
https://github.com/metasim/gdal/pull/1

Reply:
> it seems to copy what it needs out of papszArgv.

I was attempting to avoid a copy, since it's possible (at least in my use case) to construct the options and rn them over a bunch of files. But I guess a copy is in the performance noise.

Reply:
Not sure I understand. Even after https://github.com/metasim/gdal/pull/1/commits/f39e73d83c3b084e633e9f86f434c2b073e0c31a, it's still possible to reuse a `GdalDEMProcessingOptions`. The copy is made by `GDALDEMProcessingOptionsNew`, and I'm saying that we don't need to keep a borrow to the `CslStringList`.

Reply:
Yes, you're correct. Sorry, I missed the implications of where the copy was happening in the C++ code.